agents/alm.py

```diff
@@ -79,11 +76,14 @@ def get_lower_bound(self, state_batch, action_batch):
             z_batch = self.encoder_target(state_batch).sample()
             z_seq, action_seq = self._rollout_evaluation(z_batch, action_batch, std=0.1)
 
+            reward = self.reward(z_seq[:-1], action_seq[:-1])
             kl_reward = self.classifier.get_reward(z_seq[:-1], action_seq[:-1], z_seq[1:])
             discount = self.gamma * torch.ones_like(reward)
-            q_values_1, q_values_2, reward = self.critic(z_seq[-1], action_seq[-1])
+            if self.critic_mode == "model":
+                q_values_1, q_values_2, reward  = self.critic(z_seq[-1], action_seq[-1])
+            else: 
```

Comment:
It is generally a better idea to check all values explicitly and raise errors from the else branch. This way you will not be misled if you make a typo in config